pkg/operator/k8sutil/node_test.go

```diff
@@ -446,3 +446,39 @@ func TestTopologyLabels(t *testing.T) {
 	topology = ExtractTopologyFromLabels(nodeLabels, additionalTopologyLabels)
 	assert.Equal(t, 0, len(topology))
 }
+
+func TestGetKubernetesNodes(t *testing.T) {
+	clientset := optest.New(t, 3)
+	nodes, err := GetKubernetesNodes(clientset)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, len(nodes.Items))
+}
+
+func TestGetNotReadyKubernetesNodes(t *testing.T) {
```

Comment:
I think this needs more test cases. What if there are no nodes in the cluster? What if all the nodes are not ready?

Reply:
Added the test cases